lib/sasl/sasl.js

```diff
@@ -21,12 +18,12 @@ var saslMechanism = {
  *
  * @constructor
  */
-function Sasl(mechanism) {
-  if (mechanism && !u.includes(saslMechanism, mechanism)) {
-    throw new errors.NotImplementedError(
-      'Only SASL PLAIN and ANONYMOUS are supported.');
+function Sasl(mechanism, handler) {
```

Comment:
- Can you please document the shape of the handler? 
- Should we verify for certain property names in the handler and verify whether they are of type `function`?
- What happens if someone provides an empty object `{}` for the handler? I assume it will fail later.
- Clarifying and documenting above things will help me provide better typings for sasl handler

Reply:
I'm going to add property/function checks in the public API, and i'm adding samples and docs too.